rfcs/taskforce/0002-network-utility.md

```diff
@@ -0,0 +1,203 @@
+# Network Utility Tool
+
+## Goal
+
+When spinning up a network there are a few operations that can and have been automated to decrease iteration speed on testing newer versions of the network.
+
+1. Keyset Generation (Account Creation)
+2. Genesis Ledger Creation
+3. Distribution of custom network configuration and keys
+4. Deployment of infrastructure with updated configuration
+5. Monitoring the health of the network and debuging issues / crashes
+
+This document proposes a tool that unifies these processes by collecting tools already written for these purposes and packaging them in a simple CLI and JS API.
+
+# Design
+
+## Existing System
+
+For these processes we mainly use the [testnet-keys.py](https://github.com/CodaProtocol/coda-automation/blob/master/scripts/testnet-keys.py) script in coda-automation.
+
+1. Keypair generation calls `coda advanced-generate keypair` generate new accounts
+2. Genesis ledger creation happens using custom logic inside this script
+3. Distribution of configuration uses Kubernetes secrets
+4. Deployment consists of a variety of terraform configurations targeting Google Kubernetes Engine
+5. We have a variety of charts and metrics for monitoring macro network health but debugging specific issues is often manual
+
+### Improvements
+
+While this system is effective, there are a variety of potential improvements to this process.
+
+- Keypair generation depends on the daemon
+- Genesis ledger logic could be cleaned up and made more flexible to support a wider variety of configuraitons
+- These tools could be packaged together with a JavaScript API for extending functionality
+- The internal data structures for storing these keypairs, genesis configuration and distribution are not well defined and could be more robust
+- Tools for deployment and monitoring are seperate and these could be packaged in one simple workflow that uses the same underlying functionality
+
+## Proposal
+
+All of these features should be included in a simple command line tool that also exposes a TypeScript API. This service will be written in Reason with TypeScript bindings.
```

Comment:
We'd also want to support a Reason frontend to the API so we could easily internally consume it  elsewhere (this should be free).

Reply:
Ah yes the I think we might just need to drop the Typescript support from scope for closing the taskforce but the Reason bindings should be free.